pkg/reconciler/v1alpha1/autoscaling/kpa/scaler.go

```diff
@@ -169,17 +151,69 @@ func (ks *scaler) Scale(ctx context.Context, pa *pav1alpha1.PodAutoscaler, desir
 			// Only let a revision be scaled to 0 if it's been active for at
 			// least the stable window's time.
 			if pa.Status.CanMarkInactive(config.StableWindow) {
-				return desiredScale, nil
+				return desiredScale, true
 			}
 			// Otherwise, scale down to 1 until the idle period elapses
 			desiredScale = 1
 		} else { // Active=False
 			// Don't scale-to-zero if the grace period hasn't elapsed
 			if !pa.Status.CanScaleToZero(config.ScaleToZeroGracePeriod) {
-				return desiredScale, nil
+				return desiredScale, true
 			}
 		}
 	}
+	return desiredScale, false
+}
+
+func (ks *scaler) applyScale(ctx context.Context, pa *pav1alpha1.PodAutoscaler, desiredScale int32, resource *schema.GroupResource, scl *autoscalingapi.Scale) (int32, error) {
```

Comment:
This seems unused now? Lost in rebase?

Reply:
That's bad. I forgot to call the function at the end of the `Scale` function. So this part is untested.

Reply:
I don't think it is. The end of the function contained a copy of this function, so functionality was there.

Reply:
Ah thanks you're right. Fixed.